### PR TITLE
Update build instructions

### PR DESCRIPTION
**Pull request by quarkytale**
Related to https://github.com/osrf/mbari_wec_gz/issues/95

Setting an environment variable `SDF_PATH` avoids robot_state_publisher showing errors while running the main launch file. 

Note, this does not affect the sim running by default but throws errors (like in the tagged issue) which can be confusing. 

There's an open PR in sdformat_urdf to handle mode:// paths which would be extended to parse package:// paths as well (yet to confirm that). But until then, this step is necessary to run RViz alongside the sim.

Just added it in the build instructions, shall I add it in the `SETUP_SH` in the dockerfile as well?

### CODE REVIEW

**Comment by quarkytale** [comment]
@mabelzhang This is ready for review, tested the new image, sim works without any errors on startup. Though note, starting a new terminal using ./join.bash we need to source ~/setup.bash, just setting up the mbari_wec_ws/install/setup.bash won't work.

File: docker/mbari_wec/Dockerfile [hunk 1]
```diff
@@ -119,12 +119,14 @@ RUN /bin/bash -c 'source /opt/ros/humble/setup.bash \
...
 RUN touch ${SETUP_SH} \
  && chmod 755 ${SETUP_SH} \
- && echo ". ${MBARI_WEC_WS}/install/setup.bash" >> ${SETUP_SH}
+ && echo ". ${MBARI_WEC_WS}/install/setup.bash" >> ${SETUP_SH} \
+ && echo "export SDF_PATH=\$GZ_SIM_RESOURCE_PATH" >> ${SETUP_SH}
 ENV RUN_SH /home/$USERNAME/run_simulation.bash
 RUN touch ${RUN_SH} \
  && chmod 755 ${RUN_SH} \
  && echo ". ${MBARI_WEC_WS}/install/setup.bash" >> ${RUN_SH} \
+ && echo "export SDF_PATH=\$GZ_SIM_RESOURCE_PATH" >> ${RUN_SH} \
```
Comment (mabelzhang) [inline docker/mbari_wec/Dockerfile:128]: Hmm as the run file is getting bigger, I wonder if we can just call the setup file from it, like 
```suggestion
 && echo ". ${SETUP_SH}" >> ${RUN_SH} \
```
Saves some duplication.
I haven't tested this though.

**Comment by mabelzhang** [comment]
> There's an open PR in sdformat_urdf

Could you link to that PR in this PR, so we know to update the Dockerfile when that PR is merged?

A comment above the block in the Dockerfile linking to that PR, saying the SF_PATH is not needed after that PR is merged, will also be helpful for anyone wondering what it's for in the future.

File: README.md [hunk 2]
```diff
@@ -102,6 +102,12 @@ At the moment, MBARI WEC is supported by source installation only. Use Ubuntu Ja
     export GZ_VERSION=garden
     ```
 
+1. Set `SDF_PATH` to allow `robot_state_publisher` parse the robot description from the sdformat model:
```
Comment (mabelzhang) [inline README.md:105]:
```suggestion
1. Set `SDF_PATH` to allow `robot_state_publisher` to parse the robot description from the sdformat model:
```

File: README.md [hunk 3]
```diff
@@ -102,6 +102,12 @@ At the moment, MBARI WEC is supported by source installation only. Use Ubuntu Ja
...
+
+   ```
+   export SDF_PATH=$GZ_SIM_RESOURCE_PATH
```
Comment (mabelzhang) [inline README.md:108]: Hmm, I think `$GZ_SIM_RESOURCE_PATH` is blank at this point. If they have no other Gazebo workspaces, they have to compile the workspace and run `install/setup.bash` before `$GZ_SIM_RESOURCE_PATH` has any value. So this line should be moved to the Run section, after they've sourced `install/setup.sh`.

Actually, I have a question about the Run section, why do we choose to use `setup.sh` as opposed to `setup.bash` there? Could the user be using something that's not bash? We are already asking them to source `/opt/ros/humble/setup.bash`, which assumes bash shell. I think it should be consistent across the document whether we're using .sh or .bash.

File: docker/mbari_wec/Dockerfile [hunk 4]
```diff
@@ -119,12 +119,14 @@ RUN /bin/bash -c 'source /opt/ros/humble/setup.bash \
...
  && echo "ros2 launch buoy_gazebo mbari_wec.launch.py" >> ${RUN_SH}
 
 # Start the container at a bash prompt
-ENTRYPOINT ["/bin/bash" , "-c" , "source ${MBARI_WEC_WS}/install/setup.bash && /bin/bash"]
+ENTRYPOINT ["/bin/bash" , "-c", "source $SETUP_SH && /bin/bash"]
```
Comment (mabelzhang) [inline docker/mbari_wec/Dockerfile:132]: I don't remember why, but I've found it is safer to put the variable in braces. Sometimes without braces it doesn't resolve to the right thing.
```suggestion
ENTRYPOINT ["/bin/bash" , "-c", "source ${SETUP_SH} && /bin/bash"]
```

File: docs/docs/Tutorials/Install/Install_source.md [hunk 5]
```diff
@@ -47,15 +47,21 @@ Use Ubuntu 22.04.
         export GZ_VERSION=garden
         ```
 
-4. Install ROS dependencies
+4. Set `SDF_PATH` to allow `robot_state_publisher` parse the robot description
+   from the sdformat model (place this in ~/.bashrc for convenience if rebuilding often):
+       ```
+       export SDF_PATH=$GZ_SIM_RESOURCE_PATH
```
Comment (mabelzhang) [inline docs/docs/Tutorials/Install/Install_source.md:53]: Same as comment above. I think this needs to go after sourcing `install/setup.sh` (or `install/setup.bash`)

**Review by mabelzhang — COMMENTED** [review]
The new Dockerfile worked for me. There's no more error in the linked issue, and there are a lot more printouts from `robot_state_publisher`.

File: README.md (hunk 3 above)
Comment (andermi) [reply README.md:108]: Yeah could you make a pass over the docs to make everything use `setup.bash`?

Comment (quarkytale) [reply README.md:108]: Done!

**Review by mabelzhang — APPROVED** [review]
Tested latest Dockerfile and works.
Thanks!

**Pull request merged**